lang/c++/CMakeLists.txt

```diff
@@ -21,7 +21,7 @@ cmake_minimum_required (VERSION 3.1)
 set (CMAKE_LEGACY_CYGWIN_WIN32 0)
 
 if (NOT DEFINED CMAKE_CXX_STANDARD)
-    set(CMAKE_CXX_STANDARD 11)
+    set(CMAKE_CXX_STANDARD 14)
```

Comment:
Let's jump directly to 17  - https://issues.apache.org/jira/browse/AVRO-3610

Reply:
Sure, I've updated the PR.